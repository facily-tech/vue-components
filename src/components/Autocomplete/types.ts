export interface IAutocompleteItems {
  [index: string]: string;
}
