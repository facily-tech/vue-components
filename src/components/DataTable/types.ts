export interface IDataTableHeaders {
  text: string;
  align: string;
  value: string;
}

interface IDataTableColumnsObject {
  title: string;
  subtitle: string;
}

interface IDataTableColumnsCheckbox {
  checked: boolean;
  onChange?: (event: Event) => void;
}

interface IDataTableColumns {
  [index: string]: string | IDataTableColumnsObject | IDataTableColumnsCheckbox;
}

interface IDataTableActions {
  type: 'active' | 'icon-btn' | 'link';
  color: string;
  actionKey: string;
  icon: string;
  label: string;
}

export interface IDataTableRows {
  cols: IDataTableColumns;
  actions?: IDataTableActions[];
}

export interface IDataTablePagination {
  itemsPerPage: number;
  page: number;
  totalPages: number;
}
