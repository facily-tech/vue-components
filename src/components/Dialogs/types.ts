export interface IDialog {
  fullscreen: boolean;
  hideOverlay: boolean;
  scrollable: boolean;
}

export interface IDialogDeleteItem {
  id: string;
  description: string;
}
