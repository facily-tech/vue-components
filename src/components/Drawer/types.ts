export interface ISubItemsDrawer {
  link: string;
  title: string;
}

export interface IItemsDrawer {
  title: string;
  icon: string;
  enable: boolean;
  id: string;
  subItems: ISubItemsDrawer[];
  link?: string;
  divider?: boolean;
}
