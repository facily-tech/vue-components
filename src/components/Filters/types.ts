export interface IAllTags {
  key: number;
  value: string;
  enable: boolean;
  type: string;
}
