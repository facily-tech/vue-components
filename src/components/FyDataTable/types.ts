export interface IDataTableHeaders {
  text: string;
  align: string;
  value: string;
  sortable?: boolean;
}

interface IDataTableColumnsObject {
  title: string;
  subtitle: string;
}

interface IDataTableColumnsCheckbox {
  checked: boolean;
  onChange?: (event: Event) => void;
}

interface IDataTableColumns {
  [index: string]: string | IDataTableColumnsObject | IDataTableColumnsCheckbox;
}

interface IDataTableActions {
  type: 'active' | 'icon-btn' | 'link';
  color: string;
  actionKey: string;
  icon: string;
  label: string;
  loading?: boolean;
}

export interface IDataTableRows<T = IDataTableColumns> {
  cols: IDataTableColumns | T;
  actions?: IDataTableActions[];
}

export interface IDataTablePagination {
  itemsPerPage: number;
  page: number;
  totalPages: number;
}
