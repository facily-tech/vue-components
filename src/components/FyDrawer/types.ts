export interface IDrawerSubItems {
  link: string;
  title: string;
}

export interface IDrawerItems {
  title: string;
  icon: string;
  enable: boolean;
  id: string;
  subItems: IDrawerSubItems[];
  link?: string;
  divider?: boolean;
}
