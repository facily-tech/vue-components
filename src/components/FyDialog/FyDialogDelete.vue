<template>
  <fy-dialog
    :dialog="deleteModel"
    :config="{
      fullscreen: false,
      hideOverlay: false,
      scrollable: true,
    }"
    :size="'580px'"
  >
    <template v-slot:content>
      <v-card>
        <div class="delete-wrapper">
          <v-card tile>
            <v-toolbar flat light color="primary">
              <v-toolbar-title style="color: white">
                {{ title }}
              </v-toolbar-title>
              <v-spacer></v-spacer>
              <v-btn icon @click="close()">
                <v-icon color="#ffffff"> mdi-close </v-icon>
              </v-btn>
            </v-toolbar>
            <v-card-text class="pa-10">
              <div class="delete-ilustra d-flex justify-center pt-5">
                <v-img
                  lazy-src="@/assets/delete.svg"
                  max-height="212"
                  max-width="212"
                  src="@/assets/delete.svg"
                ></v-img>
              </div>

              <div class="dtext-center pt-10">
                <h2 class="headline">
                  Tem certeza que deseja excluir <strong>"{{ item.description }}"</strong>?
                </h2>
              </div>

              <div class="justify-center d-flex mt-10">
                <v-btn color="primary" :loading="loading" x-large @click="submit()">
                  Excluir
                </v-btn>
              </div>
            </v-card-text>
          </v-card>
        </div>
      </v-card>
    </template>
  </fy-dialog>
</template>

<script lang="ts">
import Vue from 'vue';

import { FyDialog } from './index';

import { IDialogDeleteItem } from './types';

export default Vue.extend({
  name: 'fy-dialog-delete',

  components: { FyDialog },

  props: {
    item: {
      type: Object as () => IDialogDeleteItem,
      default: {} as IDialogDeleteItem,
      required: true,
    },

    deleteModel: { type: Boolean, default: false, required: true },

    title: { type: String, default: 'Excluir' },
  },

  data() {
    return {
      loading: false,
    };
  },

  methods: {
    close(): void {
      this.$emit('close-dialog');
    },

    async submit(): Promise<void> {
      console.log(this.item);
      this.$emit('close-dialog');
      this.$emit('confirm-dialog', this.item);
    },
  },
});
</script>
