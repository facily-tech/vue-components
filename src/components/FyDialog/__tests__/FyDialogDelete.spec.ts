import Vue from 'vue';
import Vuetify from 'vuetify';
import FyDialogDelete from '../FyDialogDelete.vue';
import { shallowMount, Wrapper } from '@vue/test-utils';
import { IDialogDeleteItem } from '../types';

Vue.use(Vuetify);

describe('FyDialogDelete.ts', () => {
	let WRAPPER: Wrapper<any>;
	beforeEach(() => {
		WRAPPER = shallowMount(FyDialogDelete, {
			vuetify: new Vuetify(),
			propsData: {
				item: {} as IDialogDeleteItem,
				deleteModel: true,
				title: '',
				loading: false,
			},
		});
	});

	test('Component exists', () => {
		expect(WRAPPER.exists()).toBeTruthy();
	});

	test('Test close', async () => {
		WRAPPER.vm.close();
		await WRAPPER.vm.$nextTick();
		expect(WRAPPER.emitted()['close-dialog']).toBeTruthy();
	});

	test('Test submit', async () => {
		WRAPPER.vm.submit();
		await WRAPPER.vm.$nextTick();
		expect(WRAPPER.emitted()['close-dialog']).toBeTruthy();
		expect(WRAPPER.emitted()['confirm-dialog']).toBeTruthy();
	});
});
