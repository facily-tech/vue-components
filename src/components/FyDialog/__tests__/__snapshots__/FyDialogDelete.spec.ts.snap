// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyDialogDelete.ts should render component and match snapshot 1`] = `
<div class="v-dialog__container" config="[object Object]" max-width="580px">
  <!---->
</div>
`;
